lib/providers/socket/commons.js

```diff
@@ -35,6 +35,18 @@ exports.setupMethodHandler = function setupMethodHandler(emitter, params, servic
       if(typeof args[position] === 'function') {
         args.splice(position, 0, {});
       }
+
+      // If no callback was provided...
+      if (typeof args[args.length - 1] !== 'function') {
```

Comment:
It should be possible to fire back an `error` event (or something similar) like `emitter.emit('error', err)`.